src/function/localStorage.test.js

```diff
@@ -0,0 +1,51 @@
+import StoreLists from './localStorage.js';
+import TestList from './testList.js';
+
+describe('should check the item is added', () => {
+  test('should add to the list', () => {
+    const test = new TestList(1, 1, 'sleep', false);
+    const testlist = StoreLists.add(test);
+    expect(testlist).toHaveLength(1);
+  });
+
+  test('should add to the list', () => {
+    const test = new TestList(2, 2, 'finish the project', false);
+    const testlist = StoreLists.add(test);
+    expect(testlist).toHaveLength(2);
+  });
+});
+
+// describe('should delete from the array', () => {
+//   test('should delete from the list', () => {
+//     StoreLists.remove(2);
+//     expect(StoreLists.remove(2)).toHaveLength(1);
+//   });
+//   test('should delete from the list', () => {
+//     StoreLists.remove(1);
+//     expect(StoreLists.remove(1)).toHaveLength(0);
+//   });
+// });
+
+describe('should edit to the input', () => {
+  test('should edit the input', () => {
+    const test = new TestList(3, 3, 'test', false);
+    StoreLists.addList(test);
+    StoreLists.editList(3, 'meet');
+    expect(JSON.parse(localStorage.getItem('todoList'))[0].name).toBe('meet');
+  });
+});
+
+describe('update task completed status', () => {
+  test('test should change completed  status', () => {
+    const test = new TestList(2, 2, 'snack', true);
+    StoreLists.add(test);
+    expect(test.completed).toEqual(true);
+  });
+});
+
+describe('Deleteing task that are complete', () => {
+  test('test for delete checked function', () => {
+    StoreLists.clearChecked();
+    expect(JSON.parse(localStorage.getItem('todoList'))[0].completed).toBe(false);
+  });
+});
```

Comment:
- Your test to edit a task runs perfectly. You add a task to the `StoreLists`, then you call the `editList` method and pass the arguments to change the task with the index '3' to change the name to 'meet', and then you expect that inside the test function. But your other 2 methods don't follow this line. 
   - [ ] On the 'update task' test you actually don't call the `completed` method to make a change on the 'completed attribute' from false to true or from true to false. Kindly call the `completed` method on the task you added and check if the completed status is opposite to the status when the task was added to make sure that the `completed` method updates the item's 'completed' status. 
   - [ ] On the 'clear all tasks' you should check if the local storage removes all tasks with the 'completed' status with a 'true' value. Kindly change this test function to add features so we can properly check if the `clearChecked` method removes tasks as expected. You could add tasks with different 'completed' statuses, some with true, others with false, and then expect the length of the local storage to be the amount of tasks you added with the 'completed' status set to false.

Reply:
Thanks for your explanation , now i can see my mistake and tried to figure it out what would be the issues. Thanks for reviewing my code and thanks for your deep explanation.